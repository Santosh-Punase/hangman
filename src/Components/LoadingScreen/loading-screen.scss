@import 'src/assets/scss/variables';

$size: 1rem;

.hangman__loading-screen {
  width: 100%;
  height: 100vh;
  background-color: black;
  opacity: 0.5;
  display: flex;
  justify-content: center;
  align-items: center;

  .dots {
    display: block;
    width: $size;
    height: $size;
    border-radius: 50%;
    margin-right: 0.5rem;
    
    &:nth-child(1) {
      background-color: $yellow-light;
      animation: loading 1s 0.2s infinite;
    }
    &:nth-child(2) {
      background-color: $green-light;
      animation: loading 1s 0.3s infinite;
    }
    &:nth-child(3) {
      background-color: $blue-light;
      animation: loading 1s 0.6s infinite;
    }
  }
}

@keyframes loading {
  50% {
    transform: scale(1.5);
  }
  100% {
    transform: scale(0.8);
  }

}