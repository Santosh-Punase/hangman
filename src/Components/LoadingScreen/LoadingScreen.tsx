import React from 'react';

import './loading-screen.scss';

const LoadingScreen: React.FC = () => {
  
  return (
    <div className="hangman__loading-screen">
      <span className="dots"></span>
      <span className="dots"></span>
      <span className="dots"></span>
    </div>
  );
}

export { LoadingScreen };
