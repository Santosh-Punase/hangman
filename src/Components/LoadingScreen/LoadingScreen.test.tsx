import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { LoadingScreen } from './LoadingScreen';

afterEach(cleanup);

describe('Loading screen', () => {
  test('should render', () => {
    const { asFragment } = render(<LoadingScreen />);
    
    expect(asFragment()).toMatchSnapshot();
  });
});
