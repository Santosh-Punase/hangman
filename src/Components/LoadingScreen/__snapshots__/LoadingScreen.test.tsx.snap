// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading screen should render 1`] = `
<DocumentFragment>
  <div
    class="hangman__loading-screen"
  >
    <span
      class="dots"
    />
    <span
      class="dots"
    />
    <span
      class="dots"
    />
  </div>
</DocumentFragment>
`;
